src/lua/datetime.lua

```diff
@@ -864,6 +878,73 @@ local function datetime_interval_add(lhs, rhs)
     end
 end
 
+--[[
+    Parse partial ISO-8601 date string
+
+    Accepetd formats are:
+
+    Basic      Extended
+    20121224   2012-12-24   Calendar date   (ISO 8601)
+    2012359    2012-359     Ordinal date    (ISO 8601)
+    2012W521   2012-W52-1   Week date       (ISO 8601)
+    2012Q485   2012-Q4-85   Quarter date
+
+    Returns pair of constructed datetime object, and length of string
+    which has been accepted by parser.
+]]
+local function datetime_parse_date(str)
+    check_str("datetime.parse_date()")
+    local dt = ffi.new('dt_t[1]')
+    local len = tonumber(builtin.tnt_dt_parse_iso_date(str, #str, dt))
+    if len == 0 then
+        error(('invalid date format %s'):format(str), 2)
+    end
+    return datetime_new_dt(dt[0]), len
+end
+
+--[[
+    datetime parse function for strings in extended iso-8601 format
+    assumes to deal with date T time time_zone at once
+       date [T] time [ ] time_zone
+    Returns constructed datetime object.
+]]
+local function datetime_parse_full(str, tzoffset)
+    check_str(str, "datetime.parse()")
+    local date = ffi.new(datetime_t)
+    local rc = builtin.tnt_datetime_parse_full(date, str, #str, tzoffset)
+    if rc == false then
+        error(("Could not parse '%s'"):format(str))
+    end
+    return date
+end
+
+local function datetime_parse_from(str, obj)
+    check_str(str, "datetime.parse()")
+    local fmt = ''
+    local offset
+
+    if obj ~= nil then
+        check_table(obj, "datetime.parse()")
+        fmt = obj.format
+        offset = obj.tzoffset
+    end
+    check_str_or_nil(fmt, "datetime.parse()")
+
+    if offset ~= nil then
+        offset = get_timezone(offset, 'tzoffset')
+        check_range(offset, -720, 840, 'tzoffset')
+    end
+    if obj and obj.tz ~= nil then
+        nyi('tz')
+    end
+
+    if fmt == '' or fmt == 'iso8601' or fmt == 'rfc3339' then
```

Comment:
Yes, that was intentionally made as aliases (despite original RFC request). `c-dt` handles all iso8601 formats, including several relaxed extensions (and including everything requested by RFC3339 profile) _at the same function_.

There was no point to try to split implementation into strict iso8601 and strict rfc3339 because them both handled efficiently by current single implementation in `c-dt`.

Reply:
Ok, thanks for the explanation!